### ISSUE TITLE
problem installing/running on Debian (Sid)

### ISSUE THREAD

**Issue by gour** [Zettlr/Zettlr#3939]
### Description

I've migrated from Fedora to Debian (Sid) and have problem installing and running Zettlr (both stable and nightly versions).

Here is the log after attempting to install locally downloaded nightly version:

```
sudo apt install ./Zettlr-2.3.0-amd64.deb
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Note, selecting 'zettlr' instead of './Zettlr-2.3.0-amd64.deb'
The following NEW packages will be installed:
  zettlr
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/78,2 MB of archives.
After this operation, 340 MB of additional disk space will be used.
Get:1 /home/gour/Downloads/Zettlr-2.3.0-amd64.deb zettlr amd64 2.3.0 [78,2 MB]
Retrieving bug reports... Done
Parsing Found/Fixed information... Done
Selecting previously unselected package zettlr.
(Reading database ... 267421 files and directories currently installed.)
Preparing to unpack .../Zettlr-2.3.0-amd64.deb ...
Unpacking zettlr (2.3.0) ...
Setting up zettlr (2.3.0) ...
Processing triggers for desktop-file-utils (0.26-1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for gnome-menus (3.36.0-1) ...
Processing triggers for shared-mime-info (2.2-1) ...
Processing triggers for mailcap (3.70+nmu1) ...
N: Download is performed unsandboxed as root as file '/home/gour/Downloads/Zettlr-2.3.0-amd64.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)
```

After reading some reports online, I did chmod-ed file to 777, but still get the same warning/error.

I can't run it as regular user since the package is installed with 700:

```
$ ll /opt
total 0
drwxr-xr-x 1 root root  614 stu  22 01:08 Element/
drwxrwxrwx 1 root root  172 pro   4  2018 link-assistantcom/
drwxrwxr-x 1 root root  612 stu  22 09:21 Signal/
drwxr-xr-x 1 root root   26 stu  22 01:12 skypeforlinux/
drwxr-xr-x 1 root root  126 stu  22 01:00 viber/
drwx------ 1 root root  584 stu  22 11:53 Zettlr/
drwxr-xr-x 1 root root 4,6K stu  22 01:15 zoom/
```
However, attempt to run it gives:

```
sudo /opt/Zettlr/Zettlr
[1122/115443.023905:FATAL:electron_main_delegate.cc(292)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180.
fish: Job 1, 'sudo /opt/Zettlr/Zettlr' terminated by signal SIGTRAP (Trace or breakpoint trap)
```

Any hint what might be wrong with *.deb package?

This is not the only package which I install(ed) by downloading it and doing install from local folder, but for the first time I see such kind of 'sandboxed' messages...

### Reproducing

I described the steps above...

### Zettlr Version

Nightly

### Installation Method

From the Website or GitHub

### Your Platform

- [ ] Windows
- [ ] macOS
- [X] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Debian Linux, 6.0.0-4-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.0.8-1 (2022-11-11) x86_64 GNU/Linux

**Comment by gour** [comment]
I've resolved my issue with:

1. installation - by `mv`-ing downloaded package to `/tmp` and installing from there and
2. running Zettlr by `chmod`-ing `/opt/Zettlr` to 755 instead of 700 as set by default.

First problem seems to be just related to Debian/Sid, while I'm not sure about the 2nd one?

**Comment by politikundbildung** [comment]
The same problem exists in Kubuntu 22.10. `chmod -R 755 /opt/Zettlr/` solves it, even if installing the .deb from a folder different than /tmp/.

**Comment by gour** [comment]
Hello,

after some time on Fedora, I'm back on Debian/Sid and `beta2` does install cleanly as it used to in the past. :-)

**Comment by mousetrappling** [comment]
`beta2` does not install cleanly for me on Ubuntu 20.04, neither did the previous beta.  I haven't looked to see if changing the file permissions works, I've just downgraded to 2.3.0 again.

**Comment by gour** [comment]
> `beta2` does not install cleanly for me on Ubuntu 20.04, neither did the previous beta. I haven't looked to see if changing the file permissions works, I've just downgraded to 2.3.0 again.

Well, it does not install cleanly in a sense that I had to do `sudo chmod -R 755 /opt` in order to be able to launch it, but previously I was not able to install the package from my download location.

**Comment by nathanlesage** [comment]
@mousetrappling could it possibly be a problem with what https://github.com/Zettlr/Zettlr/pull/4155 fixes? Tomorrow, there'll be a new nightly, you could try that out. I didn't know that apparently one problem was that I was running the builders on a "too new" OS version.

**Comment by mousetrappling** [comment]
Maybe, except that @politikundbildung says the problem exists in Kubuntu 22.10 which is obviously much newer.

I can give the nightly build a go tomorrow and report back – am I going to be able to downgrade cleanly afterwards? I don't really want to be on the bleeding edge, I'd rather stick to betas or proper releases!

**Comment by marado** [comment]
I can confirm #4155 does *not* fix this issue.

**Comment by mousetrappling** [comment]
I've just tried the nightly which says it was built about an hour ago.  It doesn't solve the problem.

**Comment by nathanlesage** [comment]
Thanks y'all for chiming in!

First, to the question:

> I can give the nightly build a go tomorrow and report back – am I going to be able to downgrade cleanly afterwards? I don't really want to be on the bleeding edge, I'd rather stick to betas or proper releases!

You can't really downgrade automatically, but what you can do is copy the configuration folder as is to someplace else, start the newer version, try it out, and then close that one, uninstall it, install the older one, and before launching the older one, copy the config folder back.

Then, back to the issue: Apparently the installer utilizes wrong permissions. Since `apt` likely reads some config file from within the Debian package in order to apply `chmod` on the files after extraction, it could be that there is an issue with this.

There's also an issue that looks related open over on Electron Builder: https://github.com/electron-userland/electron-builder/issues/2682

(This is likely an Electron builder issue as it only affects the permissions of the extracted archive and the app performs perfectly well after the permissions are fixed).

Anyone with better experience of Debian packages/the overall package installation process here who could unzip the archive and have a look where the (at least if I'm not totally off here) wrong permissions seem to come from?

Other than that, I just checked the Electron Builder changelog, and there's nothing regarding any permission errors in there, so either it was a bug that again solved itself, or it's still a problem.

Nevertheless, I just updated Electron Builder to the most recent version today, so next Monday will be a new Nightly that then features the newer version of Electron Builder, so _maybe_ that fixes the problem, but I doubt it.

**Comment by mousetrappling** [comment]
Sorry for the delay in testing, I was a bit busy earlier in the week.

I can confirm that the new Nightly of 27 Feb does _not_ fix the problem.

I'm afraid I know nothing about the package installation process, so I can't help with looking at the package itself :/

Do you keep an archive of old nightlies?  If so, then it might be possible to track down when this changes which might help you figure out what the change was?

I have a Zettlr-2.3.0-amd64.deb that I downloaded on 10 June 2022 which installs and runs perfectly.  The original post on this bug report says that the nightly and 2.3.0 as of 22 Nov 2022 didn't work, so that gives us a known good and a known bad.  As there were two problems in the initial post it's not clear if the 2.3.0 in that case failed with this problem – I suspect not, it doesn't look like you've re-built it since June, I suspect it was just the nightly with the permissions problem.  So it feels like something changed between how the package was built in June and how it was built in November.

**Comment by mousetrappling** [comment]
The problem persists with `3.0.0-beta.4`.  I tried `sudo chmod -R 755 /opt/Zettlr/` and that did not seem to get me a runable version.  However `sudo chmod 755 /opt/Zettlr` does mean that I can run from command line but I can't run from within the gui and I can't pin it to the application bar (not sure quite what the name is), so something is still wrong.  There's a directory that seems to be missing as well, see the attached screenshots (which also show that the file permissions seem to be right it's just the directory it gets wrong).

Zettlr 2.3.0 files in /opt/Zettlr/
/home/pling/Pictures/zettlr-2.3.0-permissions.png

Zettlr 3.0.0-beta.4 files in /opt/Zettlr/
/home/pling/Pictures/zettlr-3.0.0.beta.4-permissions.png

**Comment by mousetrappling** [comment]
I'm not sure if this is related to this issue (because the installation is clearly not completely successful) or if I need to start a new issue:

When I run `beta.4` I have changed the setting so that tabs and the sidebar display the title not the filename, however that does not update until I have changed and saved the file.  If I close then restart the application then it's back to displaying filenames until I modify and save the file.

There is also something similarly strange going on with the custom css I use. If I change the custom CSS in the assets manager and save it then I get my custom CSS but not until I have done that.

I've downgraded back to 2.3.0 again as having the sidebar show me only filenames until I edit each file is basically unusable :(

**Comment by nathanlesage** [comment]
So if I look at the screenshots of the directory contents, the only difference is that the newer version doesn't contain the swiftshaders directory, but that is unlikely an indicator.

Also, it appears that this is not a completely widespread issue because some others can apparently run it fine, so it may be a problem with the specific configuration of your Debian/Ubuntu distribution, but as I have not too much experience with Desktop Linux, I'm a little bit lost here.

### If you want to help:

Since we all know that it must've happened between June 10 and November 22 last year, it would be great if someone with a Ubuntu/Debian installation who can reproduce this issue could run a `git bisect` for all the commits between the two nightlies that worked in order to identify the problematic commit.

Any volunteers? If that helps, here's what you would need to do:

1. You need the commit hash for the first commit on June 1st (which is b29248ed)
2. Clone the repo locally and switch to the `develop` branch if necessary
3. Run `git bisect start`. Then, run `git bisect bad` to indicate that the current state of the develop branch produces the faulty file. Afterward, run `git bisect good b29248ed` to indicate that the commit on June 1st still produced a working `.deb` file.
4. From now on, `git bisect` will continue to move through the various commits since then and checkout to these commits. ([Here's the `git-bisect` documentation that explains what happens](https://git-scm.com/docs/git-bisect#_basic_bisect_commands_start_bad_good))
5. After each run of `git bisect`, it will check out to various commits. You will have to run `yarn install --frozen-lockfile` after each of these checkouts, then run the corresponding commands to package the app into a `.deb` file (see the Readme file for these commands) and try out the resulting deb file in the `release` directory.
6. You then need to tell `git bisect` whether the deb file worked as expected or not. If the `.deb` file worked fine, run `git bisect good`, if it has the problems outlined here in the thread, type `git bisect bad`.
7. The bisect command will use this information to quickly home in to the one commit where the install command began to produce faulty Debian packages. Once it tells you it has found the problematic commit, report it here, and I can have a look as to what the issue could've been.
